rfcs/RFC-Uploads.md

```diff
@@ -0,0 +1,404 @@
+
+**Error Responses:**
+
+| Code | Description                                                                                                             |
+| ---- | ----------------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress upload. |
+| 400  | if the file type is invalid                                                                                            |
+
+#### POST submission/{submission_id}/upload/file
+
+An authenticated user can upload a file directly from their local machine to a data set in their submission.
+
+Asset_uuid can be included to retrieve a new access token if you were unable to complete the upload before the token
+expired. An error will be returned if the upload has already been completed. The user must own the submission and be
+the one who initiated the upload to retrieve an additional access token.
+
+If the size of the file being upload is <100MB, then a pre-signed URL will be returned.
+If the size of the file is >100MB, then an access token will be returned.
+
+**Request:**
+
+| Parameter             | Description                                                                |
+| --------------------- | -------------------------------------------------------------------------- |
+| File size             | the size of the file to be uploaded                                        |
+| File Name             | the name of the file being uploaded                                        |
+| submission_id         | identifies the submission                                                  |
+| attestation           | A boolean value indicating if the file is for attestation                  |
+| _Optional_ dataset_id | identifies the dataset being uploaded. Used if resuming an existing upload |
+
+**Response:**
+
+| Key           | Description                                                   |
+| ------------- | ------------------------------------------------------------- |
+| Access Token  | allows the user to upload their file to S3 in multiple parts. |
+| Presigned URL | allows the user to upload their file to S3 in a single part.  |
+| dataset_id    | identifies the name of the new file created.                  |
+| status     | Provides the current status of the upload.			  |
+
+**Error Responses:**
+
+| Code | Description                                                                                                             |
+| ---- | ----------------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress upload. |
+| 400  | if the file type is invalid                                                                                            |
```

Comment:
How is this determined? The RFC suggests that validation is outside the scope.

Reply:
The quick validation simply checks that the file extension is correct.

Reply:
.h5ad is the expected extension.